src/main/java/com/visenze/visearch/internal/SearchOperationsImpl.java

```diff
@@ -47,17 +70,19 @@ private PagedSearchResult uploadSearchInternal(UploadSearchParams uploadSearchPa
         String imageUrl = uploadSearchParams.getImageUrl();
         ViSearchHttpResponse response;
         if (imageFile == null && imageStream == null && (Strings.isNullOrEmpty(imageUrl))) {
-            throw new IllegalArgumentException("Must provide either an image File, InputStream of the image, or a valid image url to perform upload search");
+            throw new InternalViSearchException(ResponseMessages.INVALID_IMAGE_SOURCE);
+            // throw new IllegalArgumentException("Must provide either an image File, InputStream of the image, or a valid image url to perform upload search");
         } else if (imageFile != null) {
             try {
-                response = viSearchHttpClient.postImage("/uploadsearch", uploadSearchParams.toMap(), new FileInputStream(imageFile), imageFile.getName());
+                response = viSearchHttpClient.postImage(ENDPOINT_UPLOAD_SEARCH, uploadSearchParams.toMap(), new FileInputStream(imageFile), imageFile.getName());
             } catch (FileNotFoundException e) {
-                throw new IllegalArgumentException("Could not open the image file.", e);
+                throw new InternalViSearchException(ResponseMessages.INVALID_IMAGE_OR_URL, e);
+                // throw new IllegalArgumentException("Could not open the image file.", e);
```

Comment:
@jasonpeng yes, all use the same exception, and at the outside, the code only needed to catch one type of exception.